File: .ask/lambda/src/com/github/dannful/AniLexa/handlers/LaunchRequestHandler.java
package com.github.dannful.AniLexa.handlers;

import static com.amazon.ask.request.Predicates.requestType;

import com.amazon.ask.dispatcher.request.handler.HandlerInput;
import com.amazon.ask.dispatcher.request.handler.RequestHandler;
import com.amazon.ask.model.LaunchRequest;
import com.amazon.ask.model.Response;
import com.github.dannful.AniLexa.Constants;
import java.util.Optional;

public class LaunchRequestHandler implements RequestHandler {

  @Override
  public boolean canHandle(HandlerInput input) {
    return input.matches(requestType(LaunchRequest.class));
  }

  @Override
  public Optional<Response> handle(HandlerInput input) {
    String speechText = "Welcome to the AniLexa skill! I can assist you in managing your anime list";
    return input
      .getResponseBuilder()
      .withSpeech(speechText)
      .withSimpleCard(Constants.APP_ID, speechText)
      .withReprompt(speechText)
      .build();
  }
}

File: node/lambda/src/handlers/LaunchRequestHandler.ts
import { RequestHandler, HandlerInput } from "ask-sdk-core";
import { Response } from "ask-sdk-model";

import { APP_ID } from "..";

export const LaunchRequestHandler: RequestHandler = {
  canHandle(handlerInput: HandlerInput): boolean {
    const request = handlerInput.requestEnvelope.request;
    return request.type === "LaunchRequest";
  },
  handle(handlerInput: HandlerInput): Response {
    const speakOutput =
      "Welcome to the AniLexa skill! I can assist you in managing your anime list";
    return handlerInput.responseBuilder
      .speak(speakOutput)
      .reprompt(speakOutput)
      .withSimpleCard(APP_ID, speakOutput)
      .getResponse();
  },
};
